bitmark-references/bitmark-node-software/mining.md

```diff
@@ -0,0 +1,123 @@
+# Overview
+
+This paragraph describes mining mechanism, including hashing, communication protocol, and rewards.
+
+# Block Diagram
+
+            +--------------+  zero mq protocol   +---------------+
+            |              |  ---------------->  |               |
+            |   bitmarkd   |                     |   recorderd   |
+            |              |  <----------------  |               |
+            +--------------+  zero mq protocol   +---------------+
+
+The hashing procedure relates to `bitmarkd` and `recorderd` only. The `recorderd` is a program that performs hashing, it receives jobs from `bitmarkd` and tries to find possible hashes that meets a certain criteria, if a hash is found, the `recorderd` sends a message back to `bitmarkd` and requests a validation.
+
+When `bitmarkd` receives this message from `recorderd`, the `bitmarkd` validates the hash from this special message and returns the result to the `recorderd`.
+
+If `bitmarkd` receives no valid hashes from other nodes or from `recorderd`, then the `bitmarkd` will periodically send hash tasks to `recorderd` (currently every 1 minute), until a valid hash is found or received.
+
+## Hashing
+
+Hashing is the transformation of a string of characters into an unique value or key that represents the original string. There are different hash implementations with a wide range of algorithm, if the hash implementation is well designed, it turns to be very difficult to guess what is the original data from a given hash value.
+
+At bitmark we use `argon2` to compute the hashing, the `argon2` has some benefits such as a memory hard algorithm that provides resistance against GPU and ASIC hardware for computation as well as it provides a hash of 256 bits long.
+
+The `argon2` user-land software provides a wide range of parameters, here there is an example:
+
+```
+printf '%s' 'hello world' | argon2 'hello world' -d -l 32 -m 17 -t 4 -p 1 -r | awk '{for(i=length($1);i>0;i-=2)x=x substr($1,i-1,2);print x}'
+
+f8a17bc25cb53e848e2d09811ade4b8a037f628443661b88611faf5d9a5a1f33
+```
+
+A block hash contains the following data:
+
```

Comment:
We need to explain better what's meant by "block hash". As always, it's great to refer back to what we're already talking about in the document.

Would it be accurate to say something like: "The hash produced by `recorderd` for a Bitmark block is made from the following information, sent through the `argon2` algorithm:"

Reply:
updated